src/libopensc/card-openpgp.c

```diff
@@ -357,6 +357,25 @@ pgp_match_card(sc_card_t *card)
 }
 
 
+/* populate MF - add matching blobs listed in the pgp_objects table */
+int populate_blobs_to_mf(sc_card_t *card, struct pgp_priv_data *priv)
+{
+	pgp_do_info_t	*info;
+	for (info = priv->pgp_objects; (info != NULL) && (info->id > 0); info++) {
+		if (((info->access & READ_MASK) != READ_NEVER) && (info->get_fn != NULL)) {
+			pgp_blob_t *child = NULL;
+
+			child = pgp_new_blob(card, priv->mf, info->id, sc_file_new());
```

Comment:
If the `pgp_new_blob()` fails, the memory allocated by `sc_file_new()` is leaked (yeah, it was in the original code too but as I noticed it, it would be good to fix it).